File: app/src/main/java/tech/salroid/filmy/database/MovieProjection.java
package tech.salroid.filmy.database;

/*
 * Filmy Application for Android
 * Copyright (c) 2016 Ramankit Singh (http://github.com/webianks).
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

public class MovieProjection {

    public static final int TRENDING_MOVIE_LOADER = 1;
    public static final int INTHEATERS_MOVIE_LOADER = 2;
    public static final int UPCOMING_MOVIE_LOADER = 3;


    public static final String[] MOVIE_COLUMNS = {

            FilmContract.MoviesEntry.MOVIE_ID,
            FilmContract.MoviesEntry.MOVIE_TITLE,
            FilmContract.MoviesEntry.MOVIE_YEAR,
            FilmContract.MoviesEntry.MOVIE_POSTER_LINK

    };


    public static final String[] GET_MOVIE_COLUMNS = {

            FilmContract.MoviesEntry.MOVIE_TITLE,
            FilmContract.MoviesEntry.MOVIE_BANNER,
            FilmContract.MoviesEntry.MOVIE_DESCRIPTION,
            FilmContract.MoviesEntry.MOVIE_TAGLINE,
            FilmContract.MoviesEntry.MOVIE_TRAILER,
            FilmContract.MoviesEntry.MOVIE_RATING,
            FilmContract.MoviesEntry.MOVIE_LANGUAGE,
            FilmContract.MoviesEntry.MOVIE_RELEASED,
            FilmContract.MoviesEntry.MOVIE_CERTIFICATION,
            FilmContract.MoviesEntry.MOVIE_RUNTIME,
    };


    public static final String[] GET_SAVE_COLUMNS = {

            FilmContract.SaveEntry.SAVE_ID,
            FilmContract.SaveEntry.SAVE_TITLE,
            FilmContract.SaveEntry.SAVE_BANNER,
            FilmContract.SaveEntry.SAVE_DESCRIPTION,
            FilmContract.SaveEntry.SAVE_TAGLINE,
            FilmContract.SaveEntry.SAVE_TRAILER,
            FilmContract.SaveEntry.SAVE_RATING,
            FilmContract.SaveEntry.SAVE_LANGUAGE,
            FilmContract.SaveEntry.SAVE_RELEASED,
            FilmContract.SaveEntry._ID,
            FilmContract.SaveEntry.SAVE_YEAR,
            FilmContract.SaveEntry.SAVE_CERTIFICATION,
            FilmContract.SaveEntry.SAVE_RUNTIME,
            FilmContract.SaveEntry.SAVE_POSTER_LINK,
    };


}

File: app/src/main/java/tech/salroid/filmy/database/MovieProjection.kt
package tech.salroid.filmy.database

object MovieProjection {

    const val TRENDING_MOVIE_LOADER = 1
    const val INTHEATERS_MOVIE_LOADER = 2
    const val UPCOMING_MOVIE_LOADER = 3

    val MOVIE_COLUMNS = arrayOf(
        FilmContract.MoviesEntry.MOVIE_ID,
        FilmContract.MoviesEntry.MOVIE_TITLE,
        FilmContract.MoviesEntry.MOVIE_YEAR,
        FilmContract.MoviesEntry.MOVIE_POSTER_LINK
    )

    val GET_MOVIE_COLUMNS = arrayOf(
        FilmContract.MoviesEntry.MOVIE_TITLE,
        FilmContract.MoviesEntry.MOVIE_BANNER,
        FilmContract.MoviesEntry.MOVIE_DESCRIPTION,
        FilmContract.MoviesEntry.MOVIE_TAGLINE,
        FilmContract.MoviesEntry.MOVIE_TRAILER,
        FilmContract.MoviesEntry.MOVIE_RATING,
        FilmContract.MoviesEntry.MOVIE_LANGUAGE,
        FilmContract.MoviesEntry.MOVIE_RELEASED,
        FilmContract.MoviesEntry.MOVIE_CERTIFICATION,
        FilmContract.MoviesEntry.MOVIE_RUNTIME
    )

    val GET_SAVE_COLUMNS = arrayOf(
        FilmContract.SaveEntry.SAVE_ID,
        FilmContract.SaveEntry.SAVE_TITLE,
        FilmContract.SaveEntry.SAVE_BANNER,
        FilmContract.SaveEntry.SAVE_DESCRIPTION,
        FilmContract.SaveEntry.SAVE_TAGLINE,
        FilmContract.SaveEntry.SAVE_TRAILER,
        FilmContract.SaveEntry.SAVE_RATING,
        FilmContract.SaveEntry.SAVE_LANGUAGE,
        FilmContract.SaveEntry.SAVE_RELEASED,
        FilmContract.SaveEntry.getID(),
        FilmContract.SaveEntry.SAVE_YEAR,
        FilmContract.SaveEntry.SAVE_CERTIFICATION,
        FilmContract.SaveEntry.SAVE_RUNTIME,
        FilmContract.SaveEntry.SAVE_POSTER_LINK
    )
}